### ISSUE TITLE
RemoveNonAutomaticallyInitializedCollections executes too late

### ISSUE THREAD

**Issue by lgarczyn** [brunomikoski/ScriptableObjectCollection#160]
Setup:
* create a new collection A
* set `A.AutomaticallyLoaded` to false
* create empty scene
* add a script that calls `CollectionsRegistry.Instance.RegisterCollection(A);` on Awake
* Enter playmode

Expected results:
* collection is registered properly

Actual results:
* Awake is called, but A is still in CollectionsRegistry.collections
* RemoveNonAutomaticallyInitializedCollections is then called, removing A
* A is not loaded

Workarounds:
* wait one frame to load collections

Possible solutions:
* do not use the same collections list for editor and playmode
* use PlayModeStateChange.ExitingEditMode instead of PlayModeStateChange.EnteredEditMode

**Comment by lgarczyn** [comment]
Another issue with modifying `collections` on enter playmode:

The modifications are serialized, leading to git changes during playmode

The order is not restored properly at the end, leading to semi-random changes to the order of collections every time we leave playmode

**Comment by brunomikoski** [comment]
Hey @lgarczyn sorry about the delay to get back into it, I will update the removal of the colleciton to happen on the `ExitingEditMode` on my tests looks like it solves the issue. 

About the `git` changes, sadly its a consequence of trying to simulate what would happen on the build, I would try set the flag to do not save, but I'm afraid this could lead to even more issues.

Either way, new version should be release soon and I will update with the comment here.

**Comment by lgarczyn** [comment]
Thank you! If the flags do not save, a big risk would be the ScriptableObjects never being packaged into the build. I think the only approach would be to use Addressables.

**Comment by brunomikoski** [comment]
Yeah, let me know if the change to `ExitingEditMode` solved the issue for me, on my tests looks like was enough!
